banano-quest/banano-quest/ViewControllers/CompleteQuestViewController.swift

```diff
@@ -10,41 +10,103 @@ import UIKit
 import MapKit
 import CoreLocation
 
-class CompleteQuestViewController: UIViewController {
+class CompleteQuestViewController: UIViewController, CLLocationManagerDelegate {
 
     @IBOutlet weak var mapView: MKMapView!
-    @IBOutlet weak var numberOfBananosLabel: UILabel!
     @IBOutlet weak var bananoBackground: UIView!
-    
+    @IBOutlet weak var numberOfBananosLabel: UILabel!
+    @IBOutlet weak var distanceValueLabel: UILabel!
+    @IBOutlet weak var prizeValueLabel: UILabel!
+    @IBOutlet weak var bananoCountLabel: UILabel!
+    @IBOutlet weak var questDetailTextView: UITextView!
     @IBOutlet weak var distanceLabel: UILabel!
     @IBOutlet weak var prizeLabel: UILabel!
     
+    var locationManager = CLLocationManager()
+    var currentUserLocation: CLLocation?
+    var quest: Quest?
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         // Do any additional setup after loading the view.
+        // Map settings
+        mapView.showsUserLocation = true
+        mapView.showsCompass = true
+        
+        // Location Manager settings
+        locationManager.delegate = self
+        locationManager.desiredAccuracy = kCLLocationAccuracyNearestTenMeters
+        locationManager.requestWhenInUseAuthorization()
+        locationManager.requestAlwaysAuthorization()
+        
     }
     
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-        
+        // Background settings
         bananoBackground.layer.cornerRadius = bananoBackground.frame.size.width / 2
         bananoBackground.clipsToBounds = true
+        
+        // Checks is location services are enabled to start updating location
+        if CLLocationManager.locationServicesEnabled() {
+            locationManager.startUpdatingLocation()
+        }else{
+            print("Location services are disabled, please enable before trying again.")
```

Comment:
We should have some way to display this to the user as an error message in the base view controller?